app/models/ability.rb

```diff
@@ -0,0 +1,16 @@
+class Ability
+  include CanCan::Ability
+  def initialize(user)
+    # Define abilities for the passed in user here. For example:
+    user ||= User.new # guest user (not logged in)
+    can :read, :all
+    if user.is? :admin
+      can :manage, :all
+    else
+      can :manage, Post, author_id: user.id
+      can :manage, Comment, author_id: user.id
+      can :manage, :all
+      can :read, :all
+    end
```

Comment:
- [ ] Here, `can :manage, :all` should only apply if the role is admin. Kindly update this accordingly.